### PR TITLE
Fix mimeTypes cortex player

### PR DESCRIPTION
**Pull request by cliscum**
* Handle the promise returned by the PlayerApi and use defaults when the
  player mimeTypes call has not resolved.
* Add application/cortex-bundle to default vistar.mime_types.
* Fix invalid width/height error message.

### CODE REVIEW

**Comment by cliscum** [comment]
@jawsthegame @pbogunovich @natlownes @hkaya please review

File: src/config.coffee [hunk 1]
```diff
@@ -44,9 +45,13 @@ class Config
       @_cortexConfig['vistar.height'] = height
     else
       throw new Error "Invalid width/height: \
-        #{cortex['vistar.width']}/#{cortex['vistar.height']}"
+        #{@_cortexConfig['vistar.width']}/#{@_cortexConfig['vistar.height']}"
 
-    @_cortexConfig['vistar.mime_types'] = window.Cortex.player.mimeTypes()
+    window.Cortex.player.mimeTypes()
```
Comment (drabinowitz) [inline src/config.coffee]: is there any chance this'll produce a race condition what happens if we call `all()` before this has resolved

Comment (cliscum) [inline src/config.coffee]: You'd get the defaults in that case.

**Comment by natlownes** [comment]
+1

File: src/config.coffee [hunk 2]
```diff
@@ -10,6 +10,7 @@ defaultConfig['vistar.direct_connection'] = false
 defaultConfig['vistar.cpm_floor_cents']   = 0
 defaultConfig['vistar.ad_buffer_length']  = 8
 defaultConfig['vistar.mime_types']        = [
+  'application/cortex-bundle'
```
Comment (pbogunovich) [inline src/config.coffee]: Is the default config mime_types just for the case where the player mime_types results in an error?

File: src/config.coffee (hunk 1 above)
Comment (drabinowitz) [inline src/config.coffee]: of course right thanks, that might be an issue but hamza would probably have more perspective on that, presumably cortex-ready would only be emitted after this promise had resolved/rejected.

File: src/config.coffee (hunk 2 above)
Comment (pbogunovich) [inline src/config.coffee]: Nevermind. From the test it is obvious that this is the case.

**Comment by pbogunovich** [comment]
+2

**Comment by drabinowitz** [comment]
+2

File: src/config.coffee (hunk 1 above)
Comment (hkaya) [inline src/config.coffee]: @drabinowitz Cortex api is only available after cortex-ready is fired. This app initializes the Config object after the cortex-ready event, so eventually this call will resolve. html5-player constructs ad request bodies each time it needs to make a request. Even if we use the default values for awhile, eventually the player's mime types will get used with this implementation. I don't think there's a problem here.

**Comment by hkaya** [comment]
+2

**Pull request merged**